Return null or object for ImageAttachmentSchema response

nielslange opened the pull request:
Fixes #9908

In #9908, @whywilson reported:
> _"The return image of product categories should be null or object, but on some categories, it is an empty array. The same issues happened on the image of /wp-json/wc/store/products/categories request."_

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [ ] E2E tests

The affected unit test can be executed using the following commands:
```sh
npm run wp-env start
npm run test:php -- tests/php/StoreApi/Routes/Products.php
```

#### User Facing Testing

1. Make sure to have two test product categories. One test product category should have an image, the other one should not have an image.
2. Open `[YOUR-URL]/wp-json/wc/store/products/categories` and replace `[YOUR-URL]` with the URL of your test site.
3. Alternatively, open `[YOUR-URL]/wp-json/wc/store/products/categories/[ID]` and replace `[ID]` with the ID of the test product category.
4. Verify that the response for the test product category without image is
```json
{
	...
	"image": null,
	...
}
```
5. Verify that the response for the test product category with image is
```json
{
   ...
   "images": { ... },
   ...
}
```

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Fix: Return null or object for ImageAttachmentSchema response.

nielslange committed:
Return null or object for ImageAttachmentSchema response

mikejolley commented on src/StoreApi/Schemas/V1/ImageAttachmentSchema.php:
```diff
@@ -80,12 +80,12 @@ public function get_item_response( $attachment_id ) {
 		$thumbnail = wp_get_attachment_image_src( $attachment_id, 'woocommerce_thumbnail' );
 
-		return [
+		return (object) [
```
Make sure to update the function return type to match—it still states array.

mikejolley reviewed (changes requested):
The docblock needs changing but yes, this is what was needed.

Can we ensure that all consumers of these endpoints continue to function before merging. I don't think it will break any backwards compatibility because it would I think be treated as an object anyway, but let's be certain.

Test wise, you could test the response for the cart endpoint and ensure that images are either null or received as an object with said properties. This would be added to the PHP test suite.

nielslange committed:
Adjust docblock

nielslange commented:
> The docblock needs changing but yes, this is what was needed.

Thanks for your review, @mikejolley. I just added the `docblock` that you mentioned in your review. Somehow, my reply was added as a review. When deleting it, to reply to your review comment, it accidentally also deleted your review comment. 🙈 Nevertheless, your feedback has been adjusted.

> Can we ensure that all consumers of these endpoints continue to function before merging. I don't think it will break any backwards compatibility because it would I think be treated as an object anyway, but let's be certain.
> 
> Test wise, you could test the response for the cart endpoint and ensure that images are either null or received as an object with said properties. This would be added to the PHP test suite.

Thanks for pointing this out. I'll create additional test cases to guarantee that this change will remain backwards compatible.

nielslange committed:
Update unit tests for ImageAttachmentSchema

mikejolley reviewed (approved):
Thanks for adding the tests 🥇

opr reviewed (approved):
Looks good to me!

nielslange commented:
Thanks for your reviews, @mikejolley and @opr! 🙌
